chainstate/test-suite/benches/benches.rs

```diff
@@ -14,24 +14,88 @@
 // limitations under the License.
 
 use chainstate_test_framework::TestFramework;
-use common::primitives::Idable;
+use common::{
+    chain::{stakelock::StakePoolData, Destination, Mlt, PoolId},
+    primitives::{per_thousand::PerThousand, Amount, BlockDistance, BlockHeight, Idable, H256},
+};
+use crypto::{
+    key::{KeyKind, PrivateKey},
+    vrf::{VRFKeyKind, VRFPrivateKey},
+};
 use test_utils::random::make_seedable_rng;
 
 use criterion::{criterion_group, criterion_main, Criterion};
 
-// TODO: rework for PoS
-pub fn reorg(c: &mut Criterion) {
+pub fn pow_reorg(c: &mut Criterion) {
     let mut rng = make_seedable_rng(1111.into());
-    let mut tf = TestFramework::builder(&mut rng).build();
+    let mut tf = TestFramework::builder(&mut rng)
+        .with_chain_config(
+            common::chain::config::Builder::new(common::chain::config::ChainType::Regtest)
+                .net_upgrades(common::chain::NetUpgrades::unit_tests())
+                .max_depth_for_reorg(BlockDistance::new(5000))
+                .build(),
+        )
+        .build();
 
     let common_block_id = tf.create_chain(&tf.genesis().get_id().into(), 5, &mut rng).unwrap();
 
     tf.create_chain(&common_block_id, 1000, &mut rng).unwrap();
 
-    c.bench_function("Reorg", |b| {
-        b.iter(|| tf.create_chain(&common_block_id, 1001, &mut rng).unwrap())
+    c.bench_function("PoW reorg", |b| {
+        b.iter(|| {
+            tf.create_chain(&common_block_id, 1001, &mut rng).unwrap();
+        })
     });
 }
 
-criterion_group!(benches, reorg);
+pub fn pos_reorg(c: &mut Criterion) {
+    let mut rng = make_seedable_rng(1111.into());
+    let (staking_sk, staking_pk) = PrivateKey::new_from_rng(&mut rng, KeyKind::Secp256k1Schnorr);
+    let (vrf_sk, vrf_pk) = VRFPrivateKey::new_from_rng(&mut rng, VRFKeyKind::Schnorrkel);
+
+    let pool_id = PoolId::new(H256::random_using(&mut rng));
+    let stake_pool_data = StakePoolData::new(
+        Amount::from_atoms(40_000_000 * Mlt::ATOMS_PER_MLT),
```

Comment:
Should this use `chain_config.min_stake_pool_pledge()`?

Reply:
the problem is that ChainConfig is not constructed yet. The same with other places where I put the pool into genesis block. But I agree that it doesn't look good, will try to find a place for a constant

Reply:
Yeah, fair enough. I created a throw away in `blockprod`. Maybe wasteful, but long term better maintainable:

```
let min_stake_pool_pledge = {                    
    // throw away just to get value              
    let chain_config = create_unit_test_config();
    chain_config.min_stake_pool_pledge()         
};                                               
```

Reply:
used `create_unit_test_config ` as well